src/java/picard/util/DbSnpBitSetUtil.java

```diff
@@ -66,39 +76,56 @@ public DbSnpBitSetUtil(final File dbSnpFile, final SAMSequenceDictionary sequenc
+                           final IntervalList intervals) {
 
         if (dbSnpFile == null) throw new IllegalArgumentException("null dbSnpFile");
-        final Map<DbSnpBitSetUtil, Set<VariantType>> tmp = new HashMap<DbSnpBitSetUtil, Set<VariantType>>();
+        final Map<DbSnpBitSetUtil, Set<VariantType>> tmp = new HashMap<>();
         tmp.put(this, EnumSet.copyOf(variantsToMatch));
-        loadVcf(dbSnpFile, sequenceDictionary, tmp);
+        loadVcf(dbSnpFile, sequenceDictionary, tmp, intervals);
     }
 
     /** Factory method to create both a SNP bitmask and an indel bitmask in a single pass of the VCF. */
     public static DbSnpBitSets createSnpAndIndelBitSets(final File dbSnpFile,
                                                         final SAMSequenceDictionary sequenceDictionary) {
+        return createSnpAndIndelBitSets(dbSnpFile, sequenceDictionary, null);
+    }
+
+    /** Factory method to create both a SNP bitmask and an indel bitmask in a single pass of the VCF.
+     * If intervals are given, consider only SNP and indel sites that overlap the intervals. */
+    public static DbSnpBitSets createSnpAndIndelBitSets(final File dbSnpFile,
+                                                        final SAMSequenceDictionary sequenceDictionary,
+                                                        final IntervalList intervals) {
 
         final DbSnpBitSets sets = new DbSnpBitSets();
         sets.snps   = new DbSnpBitSetUtil();
         sets.indels = new DbSnpBitSetUtil();
 
-        final Map<DbSnpBitSetUtil, Set<VariantType>> map = new HashMap<DbSnpBitSetUtil, Set<VariantType>>();
+        final Map<DbSnpBitSetUtil, Set<VariantType>> map = new HashMap<>();
         map.put(sets.snps,   EnumSet.of(VariantType.SNP));
         map.put(sets.indels, EnumSet.of(VariantType.insertion, VariantType.deletion));
-        loadVcf(dbSnpFile, sequenceDictionary, map);
+        loadVcf(dbSnpFile, sequenceDictionary, map, intervals);
         return sets;
     }
 
     /** Private helper method to read through the VCF and create one or more bit sets. */
     private static void loadVcf(final File dbSnpFile,
                                 final SAMSequenceDictionary sequenceDictionary,
-                                final Map<DbSnpBitSetUtil, Set<VariantType>> bitSetsToVariantTypes) {
+                                final Map<DbSnpBitSetUtil, Set<VariantType>> bitSetsToVariantTypes,
+                                final IntervalList intervals) {
 
-        final VCFFileReader variantReader = new VCFFileReader(dbSnpFile);
-        final CloseableIterator<VariantContext> variantIterator = variantReader.iterator();
+        final VCFFileReader variantReader = new VCFFileReader(dbSnpFile, intervals != null);
+        final Iterator<VariantContext> variantIterator;
+        if (intervals != null) {
+            variantIterator = new ByIntervalListVariantContextIterator(variantReader, intervals);
```

Comment:
I wish I had known this existed previously!  Why isn't there are VCFFileReader.query(intervals) that creates one of these?  Sigh.